### PR TITLE
Enhance documentation readme file

### PR DESCRIPTION
**Pull request by otaviojava**
# Changes

- Change readme file

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -63,16 +63,25 @@ Whether you use built-in repository interfaces or create custom repositories, Ja
...
 [source,java]
 ----
-
 @Repository
 public interface CarRepository extends BasicRepository<Car, Long> {
 
  Page<Car> findByTypeOrderByName(CarType type, PageRequest pageRequest);
 
+ @Find
+ @OrderBy(_Car.NAME)
```
Comment (njr-11) [inline README.adoc:82]: Assuming two cars can have the same name, there needs to be additional criteria to make the position of each entity deterministic.

```suggestion
 @OrderBy(_Car.NAME)
 @OrderBy(_Car.VIN)
```

**Review by njr-11 — COMMENTED** [review]
Looks good, but I added one suggestion to ensure the example is valid.

**Review by njr-11 — APPROVED** [review]

File: README.adoc [hunk 2]
```diff
@@ -63,16 +63,26 @@ Whether you use built-in repository interfaces or create custom repositories, Ja
 
 ==== Pagination
 
-Jakarta Data also supports parameters to define pagination and sorting.
+Jakarta Data supports two types of Pagination: offset and cursor-based Pagination.
+
+- **Offset Pagination**: Offset Pagination is a popular method for efficiently managing and retrieving large datasets. It is based on dividing the dataset into pages containing a specified number of elements. This method allows developers to retrieve a subset of the dataset by identifying the page number and the maximum number of elements per page.
```
Comment (gavinking) [inline README.adoc:68]: This sounds like a definition of "pagination", not of offset-based pagination specifically.

Comment (otaviojava) [reply README.adoc]: @gavinking what do you think?

File: README.adoc [hunk 3]
```diff
@@ -65,7 +65,7 @@ Whether you use built-in repository interfaces or create custom repositories, Ja
 
 Jakarta Data supports two types of Pagination: offset and cursor-based Pagination.
 
-- **Offset Pagination**: Offset Pagination is a popular method for efficiently managing and retrieving large datasets. It is based on dividing the dataset into pages containing a specified number of elements. This method allows developers to retrieve a subset of the dataset by identifying the page number and the maximum number of elements per page.
+- **Offset Pagination**: Offset-based pagination relies on the LIMIT clause to tell the database to skip a certain number of rows before returning a specific number of rows.
```
Comment (njr-11) [inline README.adoc:68]: Users should not need prior knowledge of a LIMIT clause to understand our brief overview/introduction to pagination.  Here is a suggestion for some wording that will hopefully get the intended point across:

```suggestion
- **Offset Pagination**: Offset-based pagination computes pages relative to a positional offset from the beginning of the dataset given the requested page number and page size.
```

**Review by njr-11 — CHANGES_REQUESTED** [review]

**Review by njr-11 — APPROVED** [review]

**Pull request merged**